samples/nrf5340/netboot/README.rst

```diff
@@ -0,0 +1,82 @@
+     For details on locking, see the :ref:`fprotect_readme` driver.
+
+#. Check if there is a pending firmware upgrade of the application.
+     Invoke the :ref:`subsys_pcd` library to inspect a SRAM region shared with the application core.
+     If the application core has written an update instruction, copy the specified data range to the application partition.
+     Once the copy is done, compare the SHA of the data in the application partition against the SHA specified in the shared SRAM.
+     Communicate the result of the comparison to the application core through the shared SRAM.
+
+#. Lock the flash of the application.
+     Lock the flash that contains the application.
+     Locking is done using the ACL peripheral.
+     For details on locking, see the :ref:`fprotect_readme` driver.
+
+#. Boot the application in the network core.
+     After possibly performing a firmware update, and enabling flash protection, the network core bootloader uninitializes all peripherals that it used and boots the application.
+
+Requirements
+************
+
+  * |nRF5340DK|
+
+.. _bootloader_build_and_run:
+
+Building and running
+********************
+
+The source code of the sample can be found under :file:`samples/nrf5340/netboot/` in the |NCS| folder structure.
+
+The most common use case for the network core bootloader sample is to be included as a child image in a multi-image build, rather than being built stand-alone.
+This sample is included automatically if the application in the nrf5340 network core has the :option:`CONFIG_SECURE_BOOT` option set.
+Note that the build system will automatically enable the :option:`CONFIG_SECURE_BOOT` option for the network core application if the application on the application core has enabled :option:`CONFIG_BOOTLOADER_MCUBOOT` and :option:`CONFIG_BT_RPMSG_NRF53`.
+
+A new set of firmware update files are generated by the build system when this image is included in a multi-image build.
+These files identical to those described in :ref:`mcuboot_ncs` except that they contain the network core application firmware, and are prefixed with ``net_core_``.
+
+Testing
+=======
+
+To test the network core bootloader sample run the following commands:
+
+#. |connect_terminal|
+#. ``west build -b nrf5340dk_nrf5340_cpuapp -d build_netboot samples/bluetooth/peripheral_uart -f -- -DCONFIG_BOOTLOADER_MCUBOOT=y -Dmcuboot_CONFIG_PCD=y``
+
+   This sample (:file:`samples/bluetooth/peripheral_uart`) will automatically include the network core sample ``hci_rpmsg`` when built for the nRF5340DK.
+   Since we enable MCUBoot (``-DCONFIG_BOOTLOADER_MCUBOOT=y``) in the application core, the network core bootloader is automatically included as well.
+   In addition to this, it is necessary to enable the :ref:`subsys_pcd` subsys for the MCUBoot image (``-Dmcuboot_CONFIG_PCD=y``).
+
+#. ``west flash -d build_netboot``
+#. ``nrfjprog --program build_netboot/zephyr/net_core_app_moved_test_update.hex --sectorerase -r``
```

Comment:
Might be worth specifying that this puts the update file to the upgrade slot on the application core. At least for me this wasn't obvious until I had a look at the hex file